Overhaul site, add basic landing page

merelinguist opened the pull request.

Gunner237 commented:
Hi Dylan!

I have some reservations about the redesign, which I hope you can reassure me on.

- Currently the page appears not to link to the domain progsoc.online (missing CNAME).
- Most (if not all) of the metadata in the page appears to be unchanged from ludolatin
- There was previously a meta noindex tab (you can find reference to it in the progsocDefaults.js file) - there is no need for the site to gain a public following from the web as a whole, and would be more helpful (not least in the event of personal details being left in code by accident) for general privacy
- On a privacy note, it may be wise to keep the college name as low profile as possible, as we do not currently have any official permission from the school to appear to be affiliated (hence my removal of the college link in a previous site update).
- Although I appreciate that the old/current design is somewhat dated, I tried to code it in such a way that it is easy to understand and relatively simple (although I realise that a lot of commenting was needed in the design). The website does not necessarily function as a "storefront" for ProgSoc, rather as a convenient place to keep HTML/CSS/JavaScript projects.
- If we are to go ahead on a drastic redesign, it is important to apply this styling change to existing files, or at the very least archiving them before re-coding from scratch, rather than wiping them off the face of the earth. Certainly, some like Caesar Shift, I would like to keep working on from personal interest.

Additionally, having spent some time looking into Hack Club, although I appreciate its advantages, it may be wise to consult AF before we attempt to change how ProgSoc as a society works. One of the things that I (personally) have always liked about ProgSoc is the flexibility! Also of note is that with our small numbers, 

Sorry if this has come across as being a bit negative! I really appreciate your help on this.

Thanks!

merelinguist commented:
Right.

> Currently the page appears not to link to the domain progsoc.online (missing CNAME).

Sorry, fixed.

> Most (if not all) of the metadata in the page appears to be unchanged from ludolatin

Only one occurrence, since I copied some of the mobile-friendly stuff. Fixed.

> There was previously a meta noindex tab

I'm not sure I understand why search indexing is a problem, especially since the site is going to end up being indexed through GitHub anyway. Regardless, it is your repo, so I've added that in

> If we are to go ahead on a drastic redesign, it is important to apply this styling change to existing files, or at the very least archiving them before re-coding from scratch...

Yes, I definitely understand that. The old code will still be in GitHub, though, and I'll be restyling these and adding them back in. This commit is just a first step.

> ...having spent some time looking into Hack Club, although I appreciate its advantages, it may be wise to consult AF before we attempt to change how ProgSoc as a society works...

Definitely, I didn't really intend to make ProgSoc a member of Hack Club - those link are really just placeholders of useful articles until I add back the Caesar shift, etc

Thanks!

merelinguist commented:
Sorry, I think the CNAME was broken. Try this.
